### ISSUE TITLE
Function defined in Collection pre-request script that triggers `postman.setNextRequest` doesn't work

### ISSUE THREAD

**Issue by zzxoto** [a85/POSTMan-Chrome-Extension#9174]
**Describe the bug**
Calling the function defined in Collection pre-request script that triggers `postman.setNextRequest` doesn't work

**To Reproduce**
Have following code in Collection pre-request script:
```
function recursion() {
    postman.setNextRequest(pm.info.requestId);    
}

utils ={ recursion };
```

Have following code on Request pre-request script:
```
utils.recursion()
```

Now trigger collection runner.

**Expected behavior**
Request should be executed over and over

**Observed behavior**
Request only executed once.

**App information (please complete the following information):**
 - App Type: Native App
 - Postman Version: 7.34.0
 - OS: Windows 10

**Additional context**
- `pm.info.requestId` has the same value for collection pre-request script and for request pre-request script
- If i just call `utils.recursion()` from inside the collection pre-request script then it works

**Comment by coditva** [comment]
Hey @zzxoto, thanks for writing in!

It seems that you're declaring a variable in Collection pre-request script and trying to access it in the Request pre-request script. But Collection and Request pre-request scripts are entirely independent and have different scopes. A variable/function declared in one scope will not be present in the other.

Ref: https://learning.postman.com/docs/writing-scripts/intro-to-scripts

**Comment by zzxoto** [comment]
In the code sample posted in the original issue, since i have not prefixed the `utils` variable with `let` or `const`, it is visible outside the collection pre-request script into Request pre-request script. Indeed you can try following:
```
//colleciton pre-req script
x = 33

//request pre-req script
console.log(x) //prints 33
```

versus

```
//colleciton pre-req script
let x = 33

//request pre-req script
console.log(x) //Error variable not defined
```

**Comment by coditva** [comment]
@zzxoto You’re right. It does work now, but it’s because of a side-effect of how script execution sandbox inside Postman is implemented. We do not support or recommend using this approach because it may break in future which would lead to all collections which rely on this breaking as well.

Sorry for the confusion. Hope that helps. :)

**Comment by zzxoto** [comment]
That is a bummer then because that was my way of making helper functions and sharing downstream. How would you suggest making a reusable function in postman that I can define in one place and use it throughout my various test scripts; even better across collections. I see that there are select modules that I can require, I am not sure about defining my own though and if postman would allow that. A link to documentation or to code example would be appreciated.

**Comment by biaomingzhong** [comment]
> @zzxoto You’re right. It does work now, but it’s because of a side-effect of how script execution sandbox inside Postman is implemented. We do not support or recommend using this approach because it may break in future which would lead to all collections which rely on this breaking as well.
> 
> Sorry for the confusion. Hope that helps. :)

What about Pure Functions
